keyboards/gmmk/pro/pro.c

```diff
@@ -16,223 +16,175 @@
+    {1, CS13_SW7,  CS14_SW7,  CS15_SW7 },   //88 :          //LED7      2           7.4  0
+    {1, CS13_SW8,  CS14_SW8,  CS15_SW8 },   //89 :          //LED8      2           8.5  0
 
-    { 1, CS1_SW1, CS2_SW1, CS3_SW1 },  //  66, F12, ka5
-    { 1, CS13_SW1, CS14_SW1, CS15_SW1 },  //  67, LED, l01
-    { 1, CS16_SW1, CS17_SW1, CS18_SW1 },  //  68, LED, l11
-    { 1, CS4_SW2, CS5_SW2, CS6_SW2 },  //  69, Prt, k97
-    { 1, CS13_SW2, CS14_SW2, CS15_SW2 },  //  70, LED, l02
-    { 1, CS16_SW2, CS17_SW2, CS18_SW2 },  //  71, LED, l12
-    { 1, CS4_SW3, CS5_SW3, CS6_SW3 },  //  72, Del, k65
-    { 1, CS13_SW3, CS14_SW3, CS15_SW3 },  //  73, LED, l03
-    { 1, CS16_SW3, CS17_SW3, CS18_SW3 },  //  74, LED, l13
-    { 1, CS4_SW4, CS5_SW4, CS6_SW4 },  //  75, PgUp, k15
-    { 1, CS13_SW4, CS14_SW4, CS15_SW4 },  //  76, LED, l04
-    { 1, CS16_SW4, CS17_SW4, CS18_SW4 },  //  77, LED, l14
-    { 1, CS1_SW5, CS2_SW5, CS3_SW5 },  //  78, =, k66
-    { 1, CS10_SW5, CS11_SW5, CS12_SW5 },  //  79, Right, k05
-    { 1, CS13_SW5, CS14_SW5, CS15_SW5 },  //  80, LED, l05
-    { 1, CS16_SW5, CS17_SW5, CS18_SW5 },  //  81, LED, l15
-    { 1, CS4_SW6, CS5_SW6, CS6_SW6 },  //  82, End, k75
-    { 1, CS13_SW6, CS14_SW6, CS15_SW6 },  //  83, LED, l06
-    { 1, CS16_SW6, CS17_SW6, CS18_SW6 },  //  84, LED, l16
-    { 1, CS1_SW7, CS2_SW7, CS3_SW7 },  //  85, BSpc, ka1
-    { 1, CS4_SW7, CS5_SW7, CS6_SW7 },  //  86, PgDn, k25
-    { 1, CS13_SW7, CS14_SW7, CS15_SW7 },  //  87, LED, l07
-    { 1, CS16_SW7, CS17_SW7, CS18_SW7 },  //  88, LED, l17
-    { 1, CS1_SW8, CS2_SW8, CS3_SW8 },  //  89, ], k61
-    { 1, CS4_SW8, CS5_SW8, CS6_SW8 },  //  90, Sh_R, k91
-    { 1, CS13_SW8, CS14_SW8, CS15_SW8 },  //  91, LED, l08
-    { 1, CS16_SW8, CS17_SW8, CS18_SW8 },  //  92, LED, l18
-    { 1, CS1_SW9, CS2_SW9, CS3_SW9 },  //  93, \, ka2
-    { 1, CS4_SW9, CS5_SW9, CS6_SW9 },  //  94, Up, k35
-    { 1, CS4_SW10, CS5_SW10, CS6_SW10 },  //  95, Left, k03
-    { 1, CS1_SW11, CS2_SW11, CS3_SW11 },  //  96, Enter, ka4
-    { 1, CS4_SW11, CS5_SW11, CS6_SW11 },  //  97, Down, k73
+    {1, CS16_SW1,  CS17_SW1,  CS18_SW1 },   //90 :          //LED11     2           0.8  32
+    {1, CS16_SW2,  CS17_SW2,  CS18_SW2 },   //91 :          //LED12     2           1.9  32
+    {1, CS16_SW3,  CS17_SW3,  CS18_SW3 },   //92 :          //LED13     2           3.0  32
+    {1, CS16_SW4,  CS17_SW4,  CS18_SW4 },   //93 :          //LED14     2           4.1  32
+    {1, CS16_SW5,  CS17_SW5,  CS18_SW5 },   //94 :          //LED15     2           5.2  32
+    {1, CS16_SW6,  CS17_SW6,  CS18_SW6 },   //95 :          //LED16     2           6.3  32
+    {1, CS16_SW7,  CS17_SW7,  CS18_SW7 },   //96 :          //LED17     2           7.4  32
+    {1, CS16_SW8,  CS17_SW8,  CS18_SW8 },   //97 :          //LED18     2           8.5  32
 };
+
+//See definition in 'rgb_matrix_types.h'
+led_config_t g_led_config = { {
+    //R0      R1      R2      R3      R4      R5  R6      R7 
```

Comment:
This is drastically different from what gigahawk has.  Are you sure this is correct?

Reply:
It likely is, the difference here is that the LED indices are ordered in the `Key Matrix to LED Index` portion of `g_led_config`, which changes the order of all the LED definitions after it.

Given that the keyswitch matrix bears no resemblance to the actual layout of the keys, I had decided to make the QMK LED indices roughly match the order of the way the LED matrix is arranged according to the leaked spreadsheet from #12030 (0th LED starts in top left ESC, increasing by column first).
IMO this is much more legible (and the comments are linked to the `kXX` definitions in `pro.h`. instead of their default key, which is more useful for anyone using a non-default layout)

Reply:
Let's at least move undeglow LEDs to each other so we can use RGB indicators macro.

Reply:
I'm not sure I follow, the macro just verifies the index you pass in is in range:
```c
#define RGB_MATRIX_INDICATOR_SET_COLOR(i, r, g, b) \
    if (i >= led_min && i <= led_max) {            \
        rgb_matrix_set_color(i, r, g, b);          \
    }
```

If you want to only affect the indicators, just check that it has the underglow flag when iterating through the LEDs or hardcode the list of underglow indices.